airflow/jobs/triggerer_job_runner.py

```diff
@@ -608,6 +609,18 @@ async def run_trigger(self, trigger_id, trigger):
                 self.log.info("Trigger %s fired: %s", self.triggers[trigger_id]["name"], event)
                 self.triggers[trigger_id]["events"] += 1
                 self.events.append((trigger_id, event))
+        except asyncio.CancelledError as err:
+            trigger_timeout = trigger.task_instance.trigger_timeout
+            if trigger_timeout:
+                if not trigger_timeout.tzinfo:
+                    trigger_timeout = trigger_timeout.replace(tzinfo=timezone.utc)
+
+                if trigger_timeout < timezone.utcnow():
+                    self.log.info("Trigger cancelled due to timeout")
+                    raise asyncio.CancelledError("Trigger cancelled due to timeout")
```

Comment:
Should this re-raise the caught CancelledError instead of create a new one?

Reply:
The reason why I raise a new one is to add the message to the exception. But if the log in line 619 is enough, I think reraise makes more sense

Reply:
After a second thought, I think what you suggest makes more sense. Just updated it. Thanks!

Reply:
On my second though I’m now wondering if this should raise at all! If this `raise` is removed, the block trickles down to the `raise` below anyway (line 623); and that gives us one additional log about the exception itself.

Reply:
This (not sure whether we should add that additional log) might be why I thought of reraising 🤔 But I don't have a strong opinion on it. And I noticed it might be better to change the log level from info to error for other canceled errors as well. Let me make these changes now